components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -180,6 +178,22 @@ pub fn day_of_month(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Optio
     Ok(Some(Int::from(t.day())))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn month_name(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Bytes>> {
+    match t {
+        Some(t) => {
+            if t.is_zero() && ctx.cfg.sql_mode.contains(SqlMode::NO_ZERO_DATE) {
```

Comment:
The `DateTime` value is constructed from the `ctx`. If the value is zero and `NO_ZERO_DATE` flag is set, a warning or an error will be reported. Those warnings or errors should be eliminated by `DateTime` itself. I prefer to return `month_name` directly.

Reply:
My mistake. It has been checked here.
https://github.com/tikv/tikv/blob/58ed7e14a6f1a39c0ae50260547d78648b5f60c8/components/tidb_query/src/codec/mysql/time/mod.rs#L2153-L2162